File: src/main/java/com/itsci/mju/maebanjumpen/housekeeperskill/dto/HousekeeperSkillDTO.java
package com.itsci.mju.maebanjumpen.housekeeperskill.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.itsci.mju.maebanjumpen.skilltype.dto.SkillLevelTierDTO;
import com.itsci.mju.maebanjumpen.skilltype.dto.SkillTypeDTO;
import lombok.Data;

@Data
public class HousekeeperSkillDTO {

    @JsonProperty("skillId")
    private Integer skillId;

    // ฟิลด์ ID ที่ถูกแมปจาก HousekeeperSkillMapper
    @JsonProperty("skillLevelTierId")
    private Integer skillLevelTierId;

    @JsonProperty("housekeeperId")
    private Integer housekeeperId;

    @JsonProperty("skillTypeId")
    private Integer skillTypeId;

    @JsonProperty("skillLevelTier")
    private SkillLevelTierDTO skillLevelTier;

    @JsonProperty("skillType")
    private SkillTypeDTO skillType;

    @JsonProperty("pricePerDay")
    private Double pricePerDay;

    @JsonProperty("totalHiresCompleted")
    private Integer totalHiresCompleted;
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/housekeeperskill/dto/HousekeeperSkillDTO.kt
package com.itsci.mju.maebanjumpen.housekeeperskill.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.itsci.mju.maebanjumpen.skilltype.dto.SkillLevelTierDTO
import com.itsci.mju.maebanjumpen.skilltype.dto.SkillTypeDTO

data class HousekeeperSkillDTO(
    @JsonProperty("skillId")
    var id: Long? = 0,

    // ฟิลด์ ID ที่ถูกแมปจาก HousekeeperSkillMapper
    @JsonProperty("skillLevelTierId")
    var skillLevelTierId: Int? = null,

    @JsonProperty("housekeeperId")
    var housekeeperId: Int? = null,

    @JsonProperty("skillTypeId")
    var skillTypeId: Int? = null,

    @JsonProperty("skillLevelTier")
    var skillLevelTier: SkillLevelTierDTO? = null,

    @JsonProperty("skillType")
    var skillType: SkillTypeDTO? = null,

    @JsonProperty("pricePerDay")
    var pricePerDay: Double? = null,

    @JsonProperty("totalHiresCompleted")
    var totalHiresCompleted: Int? = null
)

